File: SimpleInterfaceWithDB/app/src/main/java/eduardosanchez/example/trabajo3/MainActivity.java
package eduardosanchez.example.trabajo3;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;


import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;


public class MainActivity extends AppCompatActivity {
    ImageButton profile1;
    ImageButton profile2;
    ImageButton profile3;
    ImageButton profile4;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView chooseP = (TextView) findViewById(R.id.textChoose);
        TextView title = (TextView) findViewById(R.id.textTitleMM);
        ImageView logo = (ImageView) findViewById(R.id.logo);
        profile1 = (ImageButton) findViewById(R.id.imageButton1);
        profile2 = (ImageButton) findViewById(R.id.imageButton2);
        profile3 = (ImageButton) findViewById(R.id.imageButton3);
        profile4 = (ImageButton) findViewById(R.id.imageButton4);
        profile1.setImageResource(R.drawable.avatar1);
        profile2.setImageResource(R.drawable.avatar2);
        profile3.setImageResource(R.drawable.avatar3);
        profile4.setImageResource(R.drawable.avatar4);
        logo.setImageResource(R.drawable.l_logos_transparent);
    }

    public void goToMenu(View view) {
        Intent menuActivity = new Intent(this, MenuActivity.class);
        startActivity(menuActivity);
    }

}

File: DiceRoller/app/src/main/java/eduardosanchez/example/diceroller/MainActivity.kt
package eduardosanchez.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView

/**
 * This activity is a simulation of the roll of a dice.
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollButton: Button = findViewById(R.id.button)
        val dice1: ImageView = findViewById(R.id.imageView)
        val dice2: ImageView = findViewById(R.id.imageView2)
        rollButton.setOnClickListener {
            rollDice(dice1)
            rollDice(dice2)
        }
        // when the app stars we make a roll so it doesn't look empty
        rollDice(dice1)
        rollDice(dice2)
    }

    /**
     * Roll the dice and update the screen
     */
    private fun rollDice(diceImage: ImageView) {
        // We make te dice and get the roll randomly
        val dice = Dice(6)
        val diceRoll = dice.roll()
        // Depending on the result we change the image of the dice
        //Find the image in the layout

        val drawableResource = when (diceRoll) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }
        diceImage.setImageResource(drawableResource)
        diceImage.contentDescription = diceRoll.toString()
    }
}
class Dice(val numSides: Int) {

    fun roll(): Int {
        return (1..numSides).random()
    }
}
